seqr/views/apis/individual_api.py

```diff
@@ -118,42 +114,101 @@ def edit_individuals_handler(request, project_guid):
+                if model.sex == expected_sex:
+                    for child in Individual.objects.filter(**{parent_id_attr: indiv_id}):
+                        updated_child = update_individuals[child.guid][1] if child.guid in update_individuals else None
+                        if not updated_child or updated_child[parent_id_key] == indiv_id:
+                            if update['sex'] != model.sex:
+                                errors.append("{} is recorded as {} and also as the {} of {}".format(indiv_id, update['sex'], parent_id_type, child.individual_id))
+                            if family_id != child.family.family_id and updated_child['family']['familyId'] != family_id:
+                                errors.append(
+                                    "{} is recorded as the {} of {} but they have different family ids: {} and {}".format(
+                                        indiv_id, parent_id_type, child.individual_id, family_id, child.family.family_id)
+                                )
+
+            # Validate parents have correct sex and family
+            if update[parent_id_key] and (update[parent_id_key] != getattr(model, parent_id_attr) or family_id != model.family.family_id):
+                parent = Individual.objects.filter(individual_id=update[parent_id_key])
+                if len(parent):
+                    parent = parent[0]
+                    if update_individuals.get(parent.guid):
+                        sex = update_individuals[parent.guid][1]['sex']
+                        parent_family_id = update_individuals[parent.guid][1]['family']['familyId']
+                    else:
+                        sex = parent.sex
+                        parent_family_id = parent.family.family_id
+
+                    if sex != expected_sex:
+                        errors.append("{} is recorded as {} and also as the {} of {}".format(update[parent_id_key], sex, parent_id_type, indiv_id))
+                    if first_check and family_id != parent_family_id:
+                        errors.append("{} is recorded as the {} of {} but they have different family ids: {} and {}".format(
+                            update[parent_id_key], parent_id_type, indiv_id, parent_family_id, family_id)
+                        )
+                else:
+                    errors.append("{} is recorded as the {} of {} but does not exist".format(update[parent_id_key], parent_id_type, indiv_id))
 
-    # TODO more validation
-    errors, warnings = validate_fam_file_records(modified_individuals_list)
     if errors:
-        return create_json_response({'errors': errors, 'warnings': warnings})
+        return create_json_response({'errors': errors}, status=400, reason='Invalid updates')
 
-    updated_families, updated_individuals = add_or_update_individuals_and_families(
-        project, modified_individuals_list)
+    individuals_by_guid = {}
+    families = []
```

Comment:
would vote for just creating families_by_guid = {} here and populating it in the loop

Reply:
I'll change this to a dict by guid here, but we need to do the actual `_get_json` call after all the updates are done so the individual list is accurate